include/seastar/http/url.hh

```diff
@@ -27,6 +27,7 @@ namespace http {
 namespace internal {
 
 bool url_decode(const std::string_view& in, sstring& out);
+sstring url_encode(const std::string_view& in);
```

Comment:
You don't explain here what is the purpose of url_encode() and the commit message is ambigous. Please keep in mind that in path?a=b&c=d you need to encode each of a, b, c, and d separately - despite the name, you cannot call url_encode once on the entire URL!